crates/ruff_linter/src/rules/flake8_bugbear/rules/duplicate_value.rs

```diff
@@ -30,29 +30,65 @@ pub struct DuplicateValue {
     value: String,
 }
 
-impl Violation for DuplicateValue {
+impl AlwaysFixableViolation for DuplicateValue {
     #[derive_message_formats]
     fn message(&self) -> String {
         let DuplicateValue { value } = self;
         format!("Sets should not contain duplicate item `{value}`")
     }
+
+    fn fix_title(&self) -> String {
+        let DuplicateValue { value } = self;
+        format!("Remove duplicate item `{value}`")
+    }
 }
 
 /// B033
-pub(crate) fn duplicate_value(checker: &mut Checker, elts: &Vec<Expr>) {
+pub(crate) fn duplicate_value(checker: &mut Checker, elts: &[Expr], range: TextRange) {
     let mut seen_values: FxHashSet<ComparableExpr> = FxHashSet::default();
-    for elt in elts {
+    let mut duplicate_indices: Vec<usize> = Vec::new();
+    let mut unique_indices: Vec<usize> = Vec::new();
+
+    for (index, elt) in elts.iter().enumerate() {
         if elt.is_literal_expr() {
             let comparable_value: ComparableExpr = elt.into();
 
-            if !seen_values.insert(comparable_value) {
-                checker.diagnostics.push(Diagnostic::new(
-                    DuplicateValue {
-                        value: checker.generator().expr(elt),
-                    },
-                    elt.range(),
-                ));
+            if seen_values.insert(comparable_value) {
+                unique_indices.push(index);
+            } else {
+                duplicate_indices.push(index);
             }
-        };
+        } else {
+            unique_indices.push(index);
+        }
+    }
+
+    if duplicate_indices.is_empty() {
+        return;
+    }
+
+    let fix = Fix::safe_edit(Edit::range_replacement(
+        checker.generator().expr(&Expr::Set(ExprSet {
+            elts: unique_indices
+                .into_iter()
+                .map(|index| elts[index].clone())
+                .collect(),
+            range: TextRange::default(),
+        })),
+        range,
+    ));
```

Comment:
I would prefer to do something similar to `remove_argument`, if possible, since the generator-based approach will lose comments. Are you open to trying that?

Reply:
Seems like `remove_argument` is strictly for `ExprCall`s _and associated `Arguments`_, unsure if it'd work to cast everything over.

All usages of `remove_argument` come from actual `ExprCall`s, and we're working with a vector of `Expr` here from a set.

Please advise! 🙏🏽

Reply:
Oh sorry, I meant, like, writing a new method that removes an element from a comma-separated collections, but based on the token stream, similar to `remove_argument`.

Reply:
Ahh. I'll see what I can do!